mmeds/documents.py

```diff
@@ -35,43 +36,69 @@ class MMEDSDoc(men.Document):
     analysis_status = men.StringField(max_length=45)
     restart_stage = men.IntField()
     pid = men.IntField()
+    exit_code = men.IntField()
     files = men.DictField()
     config = men.DictField()
 
     # When the document is updated record the
     # location of all files in a new file
-    def save(self):
-        with open(str(Path(self.path) / 'file_index.tsv'), 'w') as f:
-            f.write('{}\t{}\t{}\n'.format(self.owner, self.email, self.access_code))
-            f.write('Key\tPath\n')
-            for key, file_path in self.files.items():
-                # Skip non existent files
-                if file_path is None:
-                    continue
-                # If it's a key for an analysis point to the file index for that analysis
-                elif isinstance(file_path, dict):
-                    f.write('{}\t{}\n'.format(key, Path(self.path) / key / 'file_index.tsv'))
-                # Otherwise just write the value
-                else:
-                    f.write('{}\t{}\n'.format(key, file_path))
-        with open(DOCUMENT_LOG, 'a') as f:
-            f.write('-\t'.join([str(type(self)), self.owner, 'Upload', 'Finished', self.path, self.access_code]) + '\n')
-        super().save()
+    def save(self, **kwargs):
+        super().save(**kwargs)
+        if self.path is not None:
+            with open(str(Path(self.path) / 'file_index.tsv'), 'w') as f:
```

Comment:
A little nit-picky, but I don't like the use of f as a name for a file object. My preference would be for something more descriptive.

Reply:
In general I agree single letter variables are bad, but `f` is a standard for files in python. A bit like `i` for iterating. It's even used in the documentation. https://docs.python.org/3/tutorial/inputoutput.html That being said, I want to replaces this all with a yaml file, so there will be many fewer lines of code where `f` is defined

Reply:
I think it depends on how ambiguous ```f``` is in the given context. For instance, if you had a function that just opened a file, I think using f to represent a generic file makes sense. By contrast, your ```f``` file here seems to reference a specific file ```../../file_index.tsv``` which is referenced a few times. However, if you're going to change the code down the road anyway then I think it's fine.